kie-server-parent/kie-server-services/kie-server-services-common/src/main/java/org/kie/server/services/impl/controller/DefaultRestControllerImpl.java

```diff
@@ -146,16 +146,21 @@ public KieServerSetup connect(KieServerInfo serverInfo) {
 
         KieServerConfig config = currentState.getConfiguration();
         if (controllers != null && !controllers.isEmpty()) {
+            KieServerSetup kieServerSetup = null;
             for (String controllerUrl : controllers) {
 
                 if (controllerUrl != null && !controllerUrl.isEmpty()) {
-                    KieServerSetup kieServerSetup = connectToSingleController(serverInfo, config, controllerUrl);
-                    if (kieServerSetup != null && kieServerSetup.hasNoErrors()) {
-                        return kieServerSetup;
+                    KieServerSetup kieServerSetupController = connectToSingleController(serverInfo, config, controllerUrl);
+                    if (kieServerSetup == null && kieServerSetupController != null && kieServerSetupController.hasNoErrors()) {
+                        kieServerSetup = kieServerSetupController;
```

Comment:
With this change, it will try to connect to all controllers, returning the first succeeded one. In other implementations like `WebSocketKieServerControllerImpl` it returns with the first connected without errors, should it also be changed?

Reply:
not really. websockets has a different use case in this regard as it is a connected oriented protocol. So that would require another jira.